tests/e2e/README.md

```diff
@@ -0,0 +1,142 @@
+limitations under the License.
+-->
+
+# NVIDIA Container Toolkit – End‑to‑End (E2E) Test Suite
+
+---
+
+## 1  Scope & Goals
+This folder contains a **Ginkgo v2 / Gomega** test harness that exercises an
+NVIDIA Container Toolkit (CTK) installation on a **remote GPU‑enabled host** via
+SSH.  The suite validates that:
+
+1. CTK can be installed (or upgraded) head‑less (`INSTALL_CTK=true`).
+2. The specified **container image** runs successfully under `nvidia-container-runtime`.
+3. Errors and diagnostics are captured for post‑mortem analysis.
+
+The tests are intended for continuous‑integration pipelines, nightly
+compatibility runs, and pre‑release validation of new CTK builds.
+
+---
+
+## 2  Execution model
+* The framework **does not** spin up a Kubernetes cluster; it drives a single
+  host reachable over SSH.
+* All commands run in a Ginkgo‑managed context (`ctx`) so they abort cleanly on
+  timeout or Ctrl‑C.
+* Environment discovery happens once in `TestMain` → `getTestEnv()`; parameters
+  are therefore immutable for the duration of the run.
+
+---
+
+## 3  Prerequisites
+
+| Item | Version / requirement |
+|------|-----------------------|
+| **Go toolchain** | ≥ 1.22 (for building Ginkgo helper binaries) |
+| **GPU‑enabled Linux host** | Running a supported NVIDIA driver; reachable via SSH |
+| **SSH connectivity** | Public‑key authentication *without* pass‑phrase for unattended CI |
+| **Local OS** | Linux/macOS; POSIX shell required by the Makefile |
+
+---
+
+## 4  Environment variables
+
+| Variable | Required | Example | Description |
+|----------|----------|---------|-------------|
+| `E2E_INSTALL_CTK` | ✖ | `true` | When `true` the test installs CTK on the remote host before running the image. When `false` it assumes CTK is already present. |
+| `E2E_IMAGE_REPO` | ✔ | `ghcr.io/nvidia/container-toolkit` | Container Toolkit Image  |
+| `E2E_IMAGE_TAG` | ✔ | `latest` | Image tag |
+| `E2E_SSH_KEY` | ✔ | `/home/ci/.ssh/id_rsa` | Private key used for authentication. |
+| `E2E_SSH_USER` | ✔ | `ubuntu` | Username on the remote host. |
+| `E2E_SSH_HOST` | ✔ | `10.0.0.0` | Hostname or IP address of the target node. |
+| `E2E_SSH_PORT` | ✔ | `22` | SSH port of the target node. |
+
+> All variables are validated at start‑up; the suite aborts early with a clear
+> message if any are missing or ill‑formed.
+
+---
+
+## 5  Build helper binaries
+
+Install the latest Ginkgo CLI locally so that the Makefile can invoke it:
+
+```bash
+make ginkgo  # installs ./bin/ginkgo
+```
+
+The Makefile entry mirrors the pattern used in other NVIDIA E2E suites:
+
+```make
+bin/ginkgo:
+	GOBIN=$(CURDIR)/bin go install github.com/onsi/ginkgo/v2/ginkgo@latest
+```
+
+---
+
+## 6  Running the suite
+
+### 6.1  Basic invocation
+```bash
+INSTALL_CTK=true \
+E2E_IMAGE_REPO=ghcr.io/nvidia/container-toolkit \
+E2E_IMAGE_TAG=<image-tag> \
+SSH_KEY=$HOME/.ssh/id_rsa \
+SSH_USER=ubuntu \
+REMOTE_HOST=10.0.0.15 \
+REMOTE_PORT=22 \
```

Comment:
This is not consistent with the actual implementation.

Reply:
removed this section